app/models/school.rb

```diff
@@ -2,6 +2,8 @@
 
 class School < ApplicationRecord
   has_many :classes, class_name: :SchoolClass, inverse_of: :school, dependent: :destroy
+  has_many :lessons, dependent: :nullify
+  has_many :projects, dependent: :nullify
```

Comment:
❓ dependency between schools, classes, lessons, projects all currently have `dependent: :nullify` - ensure this is the correct behaviour and in line with the agreed data retention approach